cpp/src/c_api/graph_sg.cpp

```diff
@@ -263,6 +263,34 @@ struct create_graph_functor : public cugraph::c_api::abstract_functor {
       std::optional<rmm::device_uvector<edge_time_t>> dummy_start_times{std::nullopt};
       std::optional<rmm::device_uvector<edge_time_t>> dummy_end_times{std::nullopt};
 
+      rmm::device_uvector<vertex_t> vertices(edgelist_srcs.size(), handle_.get_stream());
+
+      if (!renumber_) {
+        raft::copy<vertex_t>(
+          vertices.data(), edgelist_srcs.data(), edgelist_srcs.size(), handle_.get_stream());
+
+        cugraph::detail::sort_ints(handle_.get_stream(),
+                                   raft::device_span<vertex_t>{vertices.data(), vertices.size()});
+
+        size_t unique_vertices_size = cugraph::detail::unique_ints(
+          handle_.get_stream(), raft::device_span<vertex_t>{vertices.data(), vertices.size()});
+
+        vertices.resize(unique_vertices_size + edgelist_dsts.size(), handle_.get_stream());
+
+        raft::copy<vertex_t>(vertices.data() + unique_vertices_size,
+                             edgelist_dsts.data(),
+                             edgelist_dsts.size(),
+                             handle_.get_stream());
+
+        cugraph::detail::sort_ints(handle_.get_stream(),
+                                   raft::device_span<vertex_t>{vertices.data(), vertices.size()});
+
+        unique_vertices_size = cugraph::detail::unique_ints(
+          handle_.get_stream(), raft::device_span<vertex_t>{vertices.data(), vertices.size()});
+
+        vertices.resize(unique_vertices_size, handle_.get_stream());
```

Comment:
To reduce memory footprint, you can sort & unique sources & destinations separately and merge instead of sort & unique sources, then copy all the destinations, and sort & unique.

You may further cut the memory footprint by first performing hash based group by (https://github.com/rapidsai/cugraph/blob/branch-25.04/cpp/include/cugraph/utilities/shuffle_comm.cuh#L761 with mem_frugal_threshold set), but this might be little bit of an overkill and you may defer this till this actually becomes a bottlenck (especially considering that renumber = false is here mainly for historical/debugging reasons).

If you just want to put working simple implementation here (as this won't be a case actual users will heavily use), just copy all sources & destinations to a single array and sort & unique.

Reply:
And don't forget to call shrink_to_fit() to actually release memory after resize.

Reply:
> To reduce memory footprint, you can sort & unique sources & destinations separately and

Right. I now sort & unique sources & destinations separately. I also call `shrink_to_fit`.

Reply:
> You may further cut the memory footprint by first performing hash based group by ...

Right, we can add this optimization if this becomes a bottleneck which might not be for now. And as you mention, the case this code is targeting is for historical/debugging reasons